public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx

```diff
@@ -7,26 +8,75 @@ import {
   onUpdateDatasourceJsonDataOption,
   updateDatasourcePluginJsonDataOption,
+  SelectableValue,
+  updateDatasourcePluginOption,
 } from '@grafana/data';
-import { Input, InlineField } from '@grafana/ui';
+import { getBackendSrv } from '@grafana/runtime';
+import { Input, InlineField, MultiSelect } from '@grafana/ui';
 import { notifyApp } from 'app/core/actions';
-import { createWarningNotification } from 'app/core/copy/appNotification';
+import { createErrorNotification, createWarningNotification } from 'app/core/copy/appNotification';
 import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
-import { store } from 'app/store/store';
+import { dispatch, store } from 'app/store/store';
 
 import { CloudWatchDatasource } from '../datasource';
 import { CloudWatchJsonData, CloudWatchSecureJsonData } from '../types';
+import { toOption } from '../utils/utils';
 
+import { MAX_LOG_GROUPS, MAX_VISIBLE_LOG_GROUPS } from './LogsQueryField';
 import { XrayLinkConfig } from './XrayLinkConfig';
 
 export type Props = DataSourcePluginOptionsEditorProps<CloudWatchJsonData, CloudWatchSecureJsonData>;
 
 export const ConfigEditor: FC<Props> = (props: Props) => {
   const { options } = props;
+  const { defaultLogGroups, logsTimeout } = options.jsonData;
 
-  const datasource = useDatasource(options.name);
+  const { datasource, setSaved } = useDatasource(options.name);
   useAuthenticationWarning(options.jsonData);
-  const logsTimeoutError = useTimoutValidation(props.options.jsonData.logsTimeout);
+  const logsTimeoutError = useTimoutValidation(logsTimeout);
+  const [logGroups, setLogGroups] = useState<SelectableValue[]>([]);
+  const [loadingLogGroups, setLoadingLogGroups] = useState(false);
+
+  const saveOptions = async (): Promise<void> => {
+    await getBackendSrv()
+      .put(`/api/datasources/${options.id}`, options)
+      .then((result: { datasource: any }) => {
+        updateDatasourcePluginOption(props, 'version', result.datasource.version);
+      });
+    setSaved(true);
+  };
+
+  const loadLogGroups = async () => {
+    await saveOptions();
+
+    // Don't call describeLogGroups if datasource or region doesn't exist
+    if (!datasource || !datasource.getActualRegion()) {
+      const missingConfig = !datasource ? 'Datasource' : 'Region';
+      dispatch(notifyApp(createErrorNotification(`Failed to get log groups: ${missingConfig} not configured`)));
+      setLogGroups([]);
+      return;
+    }
+
+    setLoadingLogGroups(true);
+    try {
+      const groups = await datasource
+        .describeLogGroups({ region: datasource.getActualRegion() })
+        .then((lg) => lg.map(toOption));
+      setLogGroups(groups);
+    } catch (err) {
+      let errMessage = 'unknown error';
+      if (typeof err !== 'string') {
+        try {
+          errMessage = JSON.stringify(err);
+        } catch (e) {}
+      } else {
+        errMessage = err;
+      }
+      dispatch(notifyApp(createErrorNotification(errMessage)));
```

Comment:
The error handling seems like a dupe of https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx#L108-L115. Can we put this in a utility or something?

Reply:
made one element for both